### PR TITLE
Fix wording for comparison operators.

### LINKED ISSUE #737: Wording for comparison operators implies complex number support for inequality operators
In https://data-apis.org/array-api/latest/API_specification/array_object.html#comparison-operators, it is stated:

>Comparison operators should be defined for arrays having any data type.

Is that right? I don't think `operator.{lt, gt, le, ge}`  should be defined for anything but real-valued datatypes. In particular, defining them for bool  is weird and for `complex*` makes completely no sense, see https://math.stackexchange.com/a/311341

In addition, `numpy.array_api` doesn't support this either:

```python
>>> xp.asarray(5 + 5j) < xp.asarray(-5 + 5j) 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/habbasi/miniforge3/envs/sparse-dev/lib/python3.11/site-packages/numpy/array_api/_array_object.py", line 631, in __lt__
    other = self._check_allowed_dtypes(other, "real numeric", "__lt__")
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/habbasi/miniforge3/envs/sparse-dev/lib/python3.11/site-packages/numpy/array_api/_array_object.py", line 142, in _check_allowed_dtypes
    raise TypeError(f"Only {dtype_category} dtypes are allowed in {op}")
TypeError: Only real numeric dtypes are allowed in __lt__
```

Note also that other operators state the datatypes for which there's actually a valid implementation.

### PR DESCRIPTION
**Pull request by hameerabbasi**
Fixes #737

### CODE REVIEW

**Comment by asmeurer** [comment]
Yes, this was a mistake. Maybe we should include the same text as in the functional version https://github.com/data-apis/array-api/blob/293512d35f8acccb55b184462505cdb1ab5d1a14/src/array_api_stubs/_2022_12/elementwise_functions.py#L1376

File: spec/2021.12/API_specification/array_object.rst [hunk 1]
```diff
@@ -170,7 +170,8 @@ A conforming implementation of the array API standard must provide and support a
     -   `operator.ne(x1, x2) <https://docs.python.org/3/library/operator.html#operator.ne>`_
     -   `operator.__ne__(x1, x2) <https://docs.python.org/3/library/operator.html#operator.__ne__>`_
 
-Comparison operators should be defined for arrays having any data type.
+:meth:`.array.__lt__`, :meth:`.array.__le__`, :meth:`.array.__gt__`, :meth:`.array.__ge__` are only defined for arrays having real-valued data types. Other comparison operators should be defined for arrays having any data type.
```
Comment (kgryte) [inline spec/2021.12/API_specification/array_object.rst:173]: This change should not be backported to the `2021.12` revision, as complex-valued dtypes were not part of the standard.

Comment (hameerabbasi) [reply spec/2021.12/API_specification/array_object.rst]: Resolved in https://github.com/data-apis/array-api/pull/736/commits/0b5e90d20d578db70809781c0bdb9ec5786783d4.

**Comment by kgryte** [comment]
@hameerabbasi Looks like there may be some trailing whitespace which needs to be removed to make CI happy.

File: spec/2022.12/API_specification/array_object.rst [hunk 2]
```diff
@@ -163,7 +163,8 @@ A conforming implementation of the array API standard must provide and support a
     -   `operator.ne(x1, x2) <https://docs.python.org/3/library/operator.html#operator.ne>`_
     -   `operator.__ne__(x1, x2) <https://docs.python.org/3/library/operator.html#operator.__ne__>`_
 
-Comparison operators should be defined for arrays having any data type.
+:meth:`.array.__lt__`, :meth:`.array.__le__`, :meth:`.array.__gt__`, :meth:`.array.__ge__` are only defined for arrays having real-valued data types. Other comparison operators should be defined for arrays having any data type.
+For backward compatibility, conforming implementations may support complex numbers; however, inequality comparison of complex numbers is unspecified and thus implementation-dependent (see :ref:`complex-number-ordering`).
```
Comment (kgryte) [inline spec/2022.12/API_specification/array_object.rst:167]:
```suggestion
For backward compatibility, conforming implementations may support complex numbers; however, inequality comparison of complex numbers is unspecified and thus implementation-dependent (see :ref:`complex-number-ordering`).
```

File: spec/draft/API_specification/array_object.rst [hunk 3]
```diff
@@ -163,7 +163,8 @@ A conforming implementation of the array API standard must provide and support a
     -   `operator.ne(x1, x2) <https://docs.python.org/3/library/operator.html#operator.ne>`_
     -   `operator.__ne__(x1, x2) <https://docs.python.org/3/library/operator.html#operator.__ne__>`_
 
-Comparison operators should be defined for arrays having any data type.
+:meth:`.array.__lt__`, :meth:`.array.__le__`, :meth:`.array.__gt__`, :meth:`.array.__ge__` are only defined for arrays having real-valued data types. Other comparison operators should be defined for arrays having any data type.
+For backward compatibility, conforming implementations may support complex numbers; however, inequality comparison of complex numbers is unspecified and thus implementation-dependent (see :ref:`complex-number-ordering`).
```
Comment (kgryte) [inline spec/draft/API_specification/array_object.rst:167]:
```suggestion
For backward compatibility, conforming implementations may support complex numbers; however, inequality comparison of complex numbers is unspecified and thus implementation-dependent (see :ref:`complex-number-ordering`).
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, Hameer!

**Pull request merged**